core/rlwe/keygenerator.go

```diff
@@ -240,36 +258,56 @@ func (kgen KeyGenerator) GenEvaluationKey(skInput, skOutput *SecretKey, evk *Eva
 	ringP := kgen.params.RingP()
 
 	// Maps the smaller key to the largest with Y = X^{N/n}.
-	ring.MapSmallDimensionToLargerDimensionNTT(skOutput.Value.Q, kgen.buffQP.Q)
+	ring.MapSmallDimensionToLargerDimensionNTT(skOutput.Value.Q, kgen.bufSkOut.Q)
 
 	// Extends the modulus P of skOutput to the one of skInput
 	if levelP := evk.LevelP(); levelP != -1 {
-		ExtendBasisSmallNormAndCenterNTTMontgomery(ringQ, ringP.AtLevel(levelP), kgen.buffQP.Q, kgen.buffQ[0], kgen.buffQP.P)
+		ExtendBasisSmallNormAndCenterNTTMontgomery(ringQ, ringP.AtLevel(levelP), kgen.bufSkOut.Q, kgen.buffQP[0].Q, kgen.bufSkOut.P)
 	}
 
 	// Maps the smaller key to the largest dimension with Y = X^{N/n}.
-	ring.MapSmallDimensionToLargerDimensionNTT(skInput.Value.Q, kgen.buffQ[0])
-	ExtendBasisSmallNormAndCenterNTTMontgomery(ringQ, ringQ.AtLevel(skOutput.Value.Q.Level()), kgen.buffQ[0], kgen.buffQ[1], kgen.buffQ[0])
+	ring.MapSmallDimensionToLargerDimensionNTT(skInput.Value.Q, kgen.bufSkIn)
+	ExtendBasisSmallNormAndCenterNTTMontgomery(ringQ, ringQ.AtLevel(skOutput.Value.Q.Level()), kgen.bufSkIn, kgen.buffQP[0].Q, kgen.bufSkIn)
 
-	kgen.genEvaluationKey(kgen.buffQ[0], kgen.buffQP, evk)
+	kgen.genEvaluationKey(kgen.bufSkIn, kgen.bufSkOut, evk)
 }
 
 func (kgen KeyGenerator) genEvaluationKey(skIn ring.Poly, skOut ringqp.Poly, evk *EvaluationKey) {
 
 	enc := kgen.WithKey(&SecretKey{Value: skOut})
 
+	// For a compressed evaluation key, a seed is created and stored in the EvaluationKey struct
+	// struct while an uncompressed key uses an ephemeral seed.
+	if evk.IsCompressed() {
+		evk.Seed = make([]byte, 32)
```

Comment:
what is the rng used for the seed?

Reply:
The seed is obtained from reading 32 bytes out of a Blake2b XOF (the PRNG contained in `kgen.Encryptor`) keyed itself with a 64 bytes `key` sampled from `crypto/rand`